lib/community/widgets/community_sidebar.dart

```diff
@@ -129,11 +136,26 @@ class _CommunitySidebarState extends State<CommunitySidebar> with TickerProvider
                                 Expanded(
                                   child: ElevatedButton(
                                     onPressed: isUserLoggedIn
-                                        ? () {
+                                        ? () async {
                                             HapticFeedback.mediumImpact();
                                             CommunityBloc communityBloc = context.read<CommunityBloc>();
                                             AccountBloc accountBloc = context.read<AccountBloc>();
                                             ThunderBloc thunderBloc = context.read<ThunderBloc>();
+
+                                            SharedPreferences prefs = (await UserPreferences.instance).sharedPreferences;
```

Comment:
It seems like this logic is mostly the same for a lot of the changes - could we abstract this a bit more to reduce code duplication? It looks like the main difference is the `draftId` but let me know if this is not possible!

Reply:
Yeah you're right there already was and now is even more duplication among all the different places where we can create posts/comments. I totally agree about trying to reduce that, although I think it might be tricky in this case. Everything defined here is needed later. For example, the draft objects are updated by the callback, the timer is canceled after the navigation push, etc. I could try to refactor the whole section (a common way to create posts/comments) but different areas use different Blocs, etc. I would be more inclined to take this as a separate task where we can re-test everything. But that's up to you!

Reply:
Sounds good - if the logic is tied to other parts, then that makes sense. We can think about a refactor of the common aspects of posts/comments in another PR